File: seminar_1/task_2/main.py
class ListNode:
	def __init__(self, x):
        	self.val = x
        	self.next = None

def reverse_list(head: ListNode) -> ListNode:
	current = head
	prev = None
	next = None
	while current:
		next = current.next
		current.next = prev
		prev = current
		current = next

	return prev

File: seminar_1/task_2/main.cpp
struct Node
{
    int     val ;
    Node*   next;
};

Node* ReverseList(Node* head)
{
    Node* cur  = head;
    Node* prev = nullptr;
    Node* next = nullptr;

    while (cur != nullptr)
    {
        next        = cur->next;
        cur->next   = prev;
        prev        = cur;
        cur         = next;
    }

    return prev;
}

int main()
{

    return 1;
}
